core/arch/aarchxx/mangle.c

```diff
@@ -2977,6 +2990,59 @@ mangle_icache_op(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
         PRE(ilist, instr, /* ldr x0, [x28] */
             instr_create_restore_from_tls(dcontext, DR_REG_X0, TLS_REG0_SLOT));
         /* Leave original instruction. */
+    } else if (opc == OP_mrs) {
+        DEBUG_DECLARE(reg_id_t src_reg = opnd_get_reg(instr_get_src(instr, 0)));
+        ASSERT(src_reg == DR_REG_CTR_EL0);
+        ptr_int_t mangle_ctr_el0 = icache_op_struct.cached_ctr_el0;
+        switch (INTERNAL_OPTION(unsafe_fake_el0_dic)) {
+        case 0:
```

Comment:
(We may want an enum for these values: but given that it's hard to share them with optionsx.h and there are only these 3 and 2 match the setting maybe it's fine -- let's leave it.)